rq/job.py

```diff
@@ -144,13 +150,66 @@ def requeue_job(job_id: str, connection: 'Redis', serializer=None) -> 'Job':
 class Job:
     """A Job is just a convenient datastructure to pass around job (meta) data."""
 
+    _dependency: Optional['Job']
     redis_job_namespace_prefix = 'rq:job:'
 
+    def __init__(self, id: Optional[str] = None, connection: Optional['Redis'] = None, serializer=None):
```

Comment:
move `__init__` up so that the type information is used elsewhere in the class, alternatively the type information can be placed like on https://github.com/rq/rq/pull/2123/files#diff-5ff734c15373fbcdb39d32faf0555265d8eaae657c1a6174e5e9b0dedd2815f7R153